File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3082_find-the-sum-of-the-power-of-all-subsequences/solutions/python3/solution.py
class Solution:
    def sumOfPower(self, nums: list[int], k: int) -> int:
        MOD = 10**9 + 7
        n = len(nums)
        # dp[j] will store the number of ways to get sum j using any number of elements
        dp = [0] * (k + 1)
        dp[0] = 1  # There's one way to make sum 0 - use no elements
        
        for num in nums:
            # Update dp array backwards for each number
            for j in range(k, num - 1, -1):
                dp[j] = (dp[j] + dp[j - num]) % MOD
        
        return dp[k]

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3082_find-the-sum-of-the-power-of-all-subsequences/solutions/java/solution.java
import java.util.Arrays;

class Solution {
    public int sumOfPower(int[] nums, int k) {
        final int MOD = 1000000007;
        int n = nums.length;
        int[] dp = new int[k + 1]; // dp[j] will be the number of ways to get sum j using any subset of nums.
        dp[0] = 1; // Base case: one way to get sum 0, use the empty set.

        for (int num : nums) {
            for (int j = k; j >= num; j--) {
                dp[j] = (dp[j] + dp[j - num]) % MOD;
            }
        }
        
        return dp[k];
    }
}
